### DISCUSSION TITLE
Add ignore unused if specifc packages are used, e.g. symfony contracts implementation as log as the interface is used

### DISCUSSION THREAD

**Discussion by AndreasA** [composer-unused/composer-unused#253] [Ideas]
Not really sure if this is the best solution but sometimes there are packages that provide interfaces and a user only develops against those interfaces.

Now if e.g. the component symfony/cache is installed but never directly used, as the code only needs the interfaces, the package is still indirectly required.

To detect if it is needed after all or not, however, can probably be a bit tricky. 

I think a quick solution would be to ignore e.g. symfony/cache as long as symfony/cache-contracts package is used. Might be a list of packages instead of just one package.

Might be related to https://github.com/composer-unused/composer-unused/issues/235

**Comment by icanhazstring** [comment]
Yep, a quick scan over packages I figured some of those are 3rd party indirect usages. As you suggested it should be solved with #235.

But I think not all packages are solved. That is why I also plan on application to provider their own symbol provider or implementations to mark packages as used. This happens starting with 0.8.x version, as I have done major rewrite
